Detectors/TPC/base/include/TPCBase/Sector.h

```diff
@@ -30,75 +30,136 @@
+
+  /// helper function to retrieve a TPC sector given cartesian coordinates
+  /// \param x x position
+  /// \param y y position
+  /// \param z z position
+  template <typename T>
+  static int ToSector(T x, T y, T z)
+  {
+    static const T invangle(static_cast<T>(180) / static_cast<T>(M_PI * 20.)); // the angle describing one sector
+    // force positive angle for conversion
+    auto s = (std::atan2(-y, -x) + static_cast<T>(M_PI)) * invangle;
+    // detect if on C size
+    if (z < static_cast<T>(0.)) {
+      s += Sector::MAXSECTOR / 2;
     }
-
-    /// int return operator to use similar as integer
-    /// \return sector number
-    operator int() const { return int(mSector); }
-
-    /// assignment operator with int
-    Sector& operator=(int sector) { mSector=sector%MAXSECTOR; return *this; }
-
-    unsigned char getSector() const { return mSector; }
-
-    Side side() const { return (mSector < MAXSECTOR / 2) ? Side::A : Side::C; }
-
-    bool looped() const { return mLoop; }
-
-    double phi() const { return (mSector % SECTORSPERSIDE) * SECPHIWIDTH + SECPHIWIDTH / 2.; }
-
-    // helper function to retrieve a TPC sector given cartesian coordinates
-    template <typename T>
-    static int ToSector(T x, T y, T z) {
-      static const T invangle(static_cast<T>(180)/static_cast<T>(M_PI*20.)); // the angle describing one sector
-      // force positive angle for conversion
-      auto s = (std::atan2(-y,-x) + static_cast<T>(M_PI))*invangle;
-      // detect if on C size
-      if (z<static_cast<T>(0.)) {
-        s += Sector::MAXSECTOR/2;
-      }
-      return s;
+    return s;
+  }
+
+  /// helper function to retrieve a TPC sector given cartesian coordinates
+  /// the sector counting is shifte by -10deg in this case, so sector 0 will
+  /// be from -10 deg to +10 deg instead of 0-20
+  /// \param x x position
+  /// \param y y position
+  /// \param z z position
+  template <typename T>
+  static int ToShiftedSector(T x, T y, T z)
+  {
+    // static int ToSector(T x, T y, T z) {
+    constexpr T invangle{ 180. / (M_PI * 20.) }; // the angle describing one sector
+    constexpr T offset{ M_PI + M_PI / 180. * 10. };
+    // force positive angle for conversion
+    auto s = static_cast<int>((std::atan2(-y, -x) + offset) * invangle) % 18;
+    // detect if on C size
+    if (z < static_cast<T>(0.)) {
+      s += Sector::MAXSECTOR / 2;
     }
+    return s;
+  }
 
-  private:
-    unsigned char mSector{};    /// Sector representation 0-MAXSECTOR
-    bool mLoop{};   /// if operator execution resulted in looping
+ private:
+  unsigned char mSector{}; /// Sector representation 0-MAXSECTOR
+  bool mLoop{};            /// if operator execution resulted in looping
 };
+
+inline Sector Sector::getLeft(const Sector s)
+{
+  const int sector = int(s);
+  const int modulus = sector % 18;
+  int offsetSector = static_cast<int>(sector / 18) * 18;
+  if (modulus == 17)
+    offsetSector -= 18;
+  return Sector(offsetSector + modulus + 1);
+}
```

Comment:
This looks rather complicated.
I guess this should only map sector to sector +1, isn't it? And it is limited to 0-17 and 18-35
So why can't you Simply use
`return Sector( ( s + 1) % 18 + (s>17)*18);`
integer conversion of `Sector` is done automatically.
It would also be good to add to the doxygen description of the function the examples that sector 17 left is 0 and 35 left is 18, or so. Just to make it clear.